bin/check-partition-usage.rb

```diff
@@ -0,0 +1,212 @@
+#! /usr/bin/env ruby
+#
+#   check-disk
+#
+# DESCRIPTION:
+#   Uses the sys-filesystem gem to get filesystem mount points disk usage
+#   --mounts '/:w75:c90,/boot:w80:c95'
+#   mounts are comma-separated
+#   warning threshold must be preceeded by the letter w and comes in the second position
+#   critical threshold must be preceeded by the letter c and comes in the third position
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux, BSD, Windows
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: sys-filesystem
+#
+# USAGE:
+#   check-partition-usage.rb --help
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2016 Magic Online - www.magic.fr
+#   Heavily inspired from check-disk-usage
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'sys/filesystem'
+include Sys
+
+#
+# Check Disk Partition Usage
+#
+class CheckDiskPartitionUsage < Sensu::Plugin::Check::CLI
+  option(
+    :mounts,
+    short: '-m MOUNTS',
+    long: '--mounts MOUNTS',
+    description: 'comma-separated mounts with their thresholds. e.g: "/boot:w80:c90"',
+    proc: proc { |a| a.split(',') },
+    default: '/boot:w80:c90'.split(',')
+  )
+
+  option(
+    :fstype,
+    short: '-t TYPE[,TYPE]',
+    description: 'Only check fs type(s)',
+    proc: proc { |a| a.split(',') }
+  )
+
+  option(
+    :ignoretype,
+    short: '-x TYPE[,TYPE]',
+    description: 'Ignore fs type(s)',
+    proc: proc { |a| a.split(',') }
+  )
+
+  option(
+    :ignoremnt,
+    short: '-i MNT[,MNT]',
+    description: 'Ignore mount point(s)',
+    proc: proc { |a| a.split(',') }
+  )
+
+  option(
+    :bwarn,
+    short: '-w PERCENT',
+    description: 'Warn if PERCENT or more of disk full',
+    proc: proc(&:to_i),
+    default: 85
+  )
+
+  option(
+    :bcrit,
+    short: '-c PERCENT',
+    description: 'Critical if PERCENT or more of disk full',
+    proc: proc(&:to_i),
+    default: 95
+  )
+
+  option(
+    :iwarn,
+    short: '-W PERCENT',
+    description: 'Warn if PERCENT or more of inodes used',
+    proc: proc(&:to_i),
+    default: 85
+  )
+
+  option(
+    :icrit,
+    short: '-K PERCENT',
```

Comment:
if you are gonna use `-W` above for the warning threshold then `-C` seems appropriate here than `-k`.